BibBuddy/app/src/main/java/de/bibbuddy/UriUtils.java

```diff
@@ -0,0 +1,355 @@
+      }
+
+    }
+
+    return null;
+  }
+
+  private static String getFileChildName(@NonNull Cursor cursor) {
+    cursor.getColumnIndex(OpenableColumns.SIZE);
+    cursor.moveToFirst();
+
+    return cursor.getString(cursor
+        .getColumnIndex(OpenableColumns.DISPLAY_NAME));
+
+  }
+
+  private static String getDataColumn(@NonNull Context context, Uri uri, String selection,
+                                      String[] selectionArgs) {
+    Cursor cursor = null;
+    final String dataColumn = UriUtilsKeys.DATA_COLUMN;
+    final String[] projection = {dataColumn};
+
+    try {
+
+      cursor = context.getContentResolver().query(uri, projection,
+          selection, selectionArgs, null);
+      if (cursor != null && cursor.moveToFirst()) {
+        return cursor.getString(cursor.getColumnIndexOrThrow(dataColumn));
+      }
+
+    } finally {
+
+      if (cursor != null) {
+        cursor.close();
+      }
+
+    }
+
+    return null;
+  }
+
+  private static String getExternalStoragePath(@NonNull Cursor cursor) {
+    String fileName = cursor.getString(0);
+    String path = Environment.getExternalStorageDirectory().toString()
+        + File.separator + StorageKeys.DOWNLOAD_FOLDER + File.separator + fileName;
+
+    if (!TextUtils.isEmpty(path)) {
+      return path;
+    }
+
+    return null;
+  }
+
+  @NonNull
+  private static String getFilePath(File file, @NonNull Context context, Uri uri) {
+
+    try {
+      InputStream inputStream = context.getContentResolver().openInputStream(uri);
+      FileOutputStream outputStream = new FileOutputStream(file);
+
+      int maxBufferSize = 1024 * 1024;
+      int bufferSize = Math.min(inputStream.available(), maxBufferSize);
+      final byte[] buffers = new byte[bufferSize];
+
+      int read;
+      while ((read = inputStream.read(buffers)) != -1) {
+        outputStream.write(buffers, 0, read);
+      }
+
+      inputStream.close();
+      outputStream.close();
+
+    } catch (IOException e) {
+      e.getMessage();
+    }
+
+    return file.getPath();
+  }
+
+  private static boolean isExternalStorageDocument(@NonNull Uri uri) {
+    return UriUtilsKeys.PROVIDER_EXTERNAL_STORAGE_DOCUMENTS
+        .equals(uri.getAuthority());
+  }
+
+  private static boolean isDownloadsDocument(@NonNull Uri uri) {
+    return UriUtilsKeys.PROVIDER_DOWNLOAD_DOCUMENTS
+        .equals(uri.getAuthority());
+  }
+
+  private static boolean isMediaDocument(@NonNull Uri uri) {
+    return UriUtilsKeys.PROVIDER_MEDIA_DOCUMENTS
+        .equals(uri.getAuthority());
+  }
+
+  private static boolean isGooglePhotosUri(@NonNull Uri uri) {
+    return UriUtilsKeys.PROVIDER_PHOTOS_CONTENT
+        .equals(uri.getAuthority());
+  }
+
+  private static boolean isGoogleDriveUri(@NonNull Uri uri) {
+    return UriUtilsKeys.AUTHORITY_DOCUMENT_STORAGE.equals(uri.getAuthority())
+            || UriUtilsKeys.AUTHORITY_DOCUMENT_STORAGE_LEGACY.equals(uri.getAuthority());
+  }
+
+}
```

Comment:
Please add a newline at the end of the file.
```suggestion
}

```